grails-app/domain/org/pih/warehouse/shipping/ShipmentItem.groovy

```diff
@@ -121,6 +121,11 @@ class ShipmentItem implements Comparable, Serializable {
         return orderNames ? orderNames.first() : ''
     }
 
+    String getPaymentTerm() {
+        List<String> orderPaymentTerms = orderItems?.collect { OrderItem orderItem -> orderItem.order.paymentTerm?.name }?.unique()
+        return orderPaymentTerms ? orderPaymentTerms.first() : ''
+    }
+
```

Comment:
Yes, that's right, it is associated with **Order**.
To generate onOrderReport it uses the **ShipmentItem** items 
`ReportController > showOnOrderReport`
```
paymentTerm  : isOrderItem ? (it.order.paymentTerm?.name ?: '') : (it.shipment.isFromPurchaseOrder ? (it?.paymentTerm ?: '') : ''),
```

so in case of when `isOrderItem = false` and `it.shipment.isFromPurchaseOrder = true` we want to extract the `Order` from the shipment.
Just like you have noticed I followed the same principal like the methods above for extracting the **Order number** and **Order Name**.

The problem is that on shipment we don;t have a direct connection to Order and so we have to dig through OrderItems inside of the shipment to extract this order